app Influx export service: custom config is not getting updated through config provider (Consul registry)

<!--🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅

Hello there! 😄

To expedite issue processing please search open and closed issues before submitting a new one.
Existing issues often contain information about workarounds, resolution, or progress updates.

Please do not remove sections of this template when submitting your issue.  If the section is not relevant to your issue, leave the section but enter "n/a" for not applicable.

Note some sections are marked "REQUIRED" and must be provided.  Never mark required sections "n/a".

🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅-->


# 🐞 Bug Report

### Affected Services [**REQUIRED**]
<!-- Can you pin-point one or more EdgeX services (Data, Metadata, Command, etc...) as the source of the bug? -->
<!-- ✍️edit: --> The issue is located in: app-influx-export https://github.com/edgexfoundry/edgex-examples/tree/main/application-services/custom/influxdb-export


### Is this a regression? No


### Description and Minimal Reproduction [**REQUIRED**]
we are currently working on containerizing the influxdb-export [app](https://github.com/edgexfoundry/edgex-examples/tree/main/application-services/custom/influxdb-export) service and run it using the docker-compose.

The service is running as expected but there is an observation that whenever we update any custom values like the mqtt broker address using the consul registry it is not reflected and if we restart the application service container then the old value from the configuration.yaml is getting overridden hence not accepting any new values through the consul UI.

Below entries are already updated on the docker-compose file
EDGEX_ADD_REGISTRY_ACL_ROLES: "app-influxdb-export"
EDGEX_ADD_KNOWN_SECRETS: redisdb[app-influxdb-export],message-bus[app-influxdb-export]
EDGEX_ADD_SECRETSTORE_TOKENS: "app-influxdb-export"

Note: No changes in the code or the configuration.yaml, The only thing is using this code a docker image is built and used as a docker container.

Any keys under the path `Key/Values/edgex/v3/app-influxdb-export/MqttConfig` is not getting updated.

I tried to change the AuthMode: "usernamepassword" to "none" and change the [BrokerAddress](https://github.com/edgexfoundry/edgex-examples/blob/77e035913dd1d869806c6913e9d730e7d3f78349/application-services/custom/influxdb-export/res/configuration.yaml#L29)

I can update the StartupMsg and also can see the new message after restarting the container

`level=INFO ts=2024-03-08T06:48:36.014365822Z app=app-influxdb-export source=service.go:244 msg="App service export to InfluxDB has Started **again**"`

## 🔥 Exception or Error
<pre><code>
<!-- If the issue is accompanied by an exception or an error, please share it below: -->
<!-- ✍️-->
2024-03-08T06:48:02.282Z [ERROR] agent.http: Request error: method=POST url=/v1/internal/acl/authorize?dc=dc1 from=192.168.160.1:35464 error="ACL not found"
2024-03-08T06:48:02.498Z [ERROR] agent.http: Request error: method=POST url=/v1/internal/acl/authorize?dc=dc1 from=192.168.160.1:35464 error="ACL not found"
2024-03-08T06:48:02.662Z [ERROR] agent.http: Request error: method=GET url=/v1/kv/edgex/v3/app-influxdb-export/MqttConfig/BrokerAddress?dc=dc1 from=192.168.160.1:35474 error="ACL not found"
</code></pre>


## 🌍  Your Environment
**Deployment Environment:** Dev

**EdgeX Version [**REQUIRED**]:** 3.1.0


**Anything else relevant?**
![image](https://github.com/edgexfoundry/app-functions-sdk-go/assets/137407921/36c924d4-4be1-4ac3-a191-6a17ad42f761)

--- a/application-services/custom/influxdb-export/main.go
+++ b/application-services/custom/influxdb-export/main.go
@@ -34,7 +34,7 @@ func main() {
 
 	// 3) Load the MQTT custom configuration
 	config := &ServiceConfig{}
-	if err := service.LoadCustomConfig(config, "MqttSecretConfig"); err != nil {
+	if err := service.LoadCustomConfig(config, "MqttConfig"); err != nil {
 		lc.Errorf("LoadCustomConfig failed: %s", err.Error())
 		os.Exit(-1)
 	}

